packages/cli/src/services/project-checker.ts

```diff
@@ -69,3 +71,66 @@ export async function checkResourceExists(name: string, placementDir: string, ex
     })
   }
 }
+
+export async function checkCurrentDirBoosterVersion(userAgent: string): Promise<void> {
+  return checkBoosterVersion(userAgent, process.cwd())
+}
+
+async function checkBoosterVersion(userAgent: string, projectPath: string): Promise<void> {
+  const projectVersion = await getBoosterVersion(projectPath)
+  const cliVersion = userAgent.split(' ')[0].split('/')[2]
+  await compareVersionsAndDisplayMessages(cliVersion, projectVersion, projectPath)
+}
+
+async function getBoosterVersion(projectPath: string): Promise<string> {
+  const projectAbsolutePath = path.resolve(projectPath)
+  try {
+    const packageJsonContents = require(path.join(projectAbsolutePath, 'package.json'))
+    const version = packageJsonContents.dependencies['@boostercloud/framework-core']
+    const versionParts = version.replace('^','').replace('.tgz','').split('-')
+    return versionParts[versionParts.length-1]    
+  } catch (e) {
+    throw new Error(
+      `There was an error when recognizing the application. Make sure you are in the root path of a Booster project:\n${e.message}`
+    )
+  }
+}
+
+async function compareVersionsAndDisplayMessages(cliVersion: string, projectVersion: string, projectPath: string): Promise<void> {
+  if (cliVersion === projectVersion)  { return }
+  const cliVersionParts = cliVersion.split('.').map((v) => parseInt(v,10))
+  const projectVersionParts = projectVersion.split('.').map((v) => parseInt(v,10))
+  if (cliVersionParts.length !== projectVersionParts.length) {
+    throw new Error(`Versions must have the same length. CLI version: ${cliVersion}. Project Booster version: ${projectVersion}`)
+  }
+  if (projectVersionParts.length !== 3) {
+    throw new Error(`Versions must follow semantic convention X.Y.Z | CLI version: ${cliVersion}. Project Booster version: ${projectVersion}`)
+  }
```

Comment:
This would go into the `parseSemver` function

Reply:
I add this logic to Semver class